packages/components/src/public/QueryModel/SaveViewModal.tsx

```diff
@@ -0,0 +1,124 @@
+        setErrorMessage(undefined);
+        setIsSubmitting(true);
+
+        try {
+            const name = isDefaultView ? '' : viewName.trim();
+            const isCurrentView = name?.toLowerCase() === currentView?.name?.toLowerCase();
+            await onConfirmSave(name, canInherit, isCurrentView);
+            afterSave(name);
+        } catch (error) {
+            setErrorMessage(resolveErrorMessage(error));
+        } finally {
+            setIsSubmitting(false);
+        }
+    }, [viewName]);
+
+    const onViewNameChange = useCallback((evt: ChangeEvent<HTMLInputElement>) => setViewName(evt.target.value), []);
+
+    const toggleDefaultView = useCallback((evt: ChangeEvent<HTMLInputElement>) => {
+        setIsDefaultView(evt.target.checked);
+    }, []);
+
+    const toggleInherit = useCallback((evt: ChangeEvent<HTMLInputElement>) => setCanInherit(evt.target.checked), []);
+
+    return (
+        <Modal onHide={onCancel} show>
+            <Modal.Header closeButton>
+                <Modal.Title>Save Grid View</Modal.Title>
+            </Modal.Header>
+            <Modal.Body>
+                <Alert>{errorMessage}</Alert>
+                <form onSubmit={saveView}>
+                    <div className="form-group">
+                        <div className="bottom-spacing">
+                            Sort order and filters are not saved as part of custom grid views. Once saved, this view
+                            will be available for all {gridLabel} grids throughout the application. Learn more about{' '}
+                            <HelpLink topic={CUSTOM_VIEW}>custom grid views</HelpLink> in LabKey.
+                        </div>
+                        <div className="bottom-spacing">
+                            <input
+                                name="gridViewName"
+                                placeholder="Grid View Name"
+                                className="form-control"
+                                value={viewName}
+                                onChange={onViewNameChange}
+                                disabled={isDefaultView}
+                                type="text"
+                            />
+                        </div>
+                        <RequiresPermission perms={PermissionTypes.Admin}>
+                            <div className="form-check bottom-spacing">
+                                <input
+                                    className="form-check-input"
+                                    type="checkbox"
+                                    name="setDefaultView"
+                                    onChange={toggleDefaultView}
+                                    checked={isDefaultView}
+                                />
+                                <span className="margin-left">Make this the default grid view</span>
+                            </div>
+                        </RequiresPermission>
+                        {isSubfolderDataEnabled() && (
```

Comment:
does this need any permissions check (in addition to the `isSubfolderDataEnabled()` check)? I'd have to check on the server side who is allowed to create an inherited view (maybe it is anyone).

Reply:
Looks like it's everyone